util.go

```diff
@@ -154,11 +155,43 @@ func trimeSchemaRefLinkPrefix(ref string) string {
 	return strings.TrimPrefix(ref, "#/components/schemas/")
 }
 
-func genSchemeaObjectID(pkgName, typeName string) string {
-	typeNameParts := strings.Split(typeName, ".")
-	pkgName = replaceBackslash(pkgName)
+func genSchemaObjectID(pkgName, typeName string, aliases map[string]*string) string {
+	validatedPkgName := checkAndMutatePackageName(pkgName, aliases)
+	validatedTypeName := checkAndMutateTypeName(typeName, aliases)
+	typeNameParts := strings.Split(validatedTypeName, ".")
+	pkgName = replaceBackslash(validatedPkgName)
 	pkgNameParts := strings.Split(pkgName, "/")
-	return strings.Join(append([]string{pkgNameParts[len(pkgNameParts)-1]}, typeNameParts[len(typeNameParts)-1]), ".")
+	if pkgNameParts[len(pkgNameParts)-1] == "" {
+		return typeNameParts[len(typeNameParts)-1]
+	} else {
+		return strings.Join(append([]string{pkgNameParts[len(pkgNameParts)-1]}, typeNameParts[len(typeNameParts)-1]), ".")
+	}
+}
+
+func checkAndMutatePackageName(pkgName string, aliases map[string]*string) string {
+	pkgNameParsed := replaceBackslash(pkgName)
+	pkgNameParts := strings.Split(pkgNameParsed, "/")
+	lastPart := pkgNameParts[len(pkgNameParts)-1]
+	if val, ok := aliases[lastPart]; ok {
+		return *val
+	} else {
+		return pkgNameParsed
+	}
+}
+
+func checkAndMutateTypeName(typeName string, aliases map[string]*string) string {
+	typeNameParsed := replaceBackslash(typeName)
```

Comment:
This may not be needed - I notice that `typeName` is never passed through `replaceBackslash` in the original code.